maven-core/src/main/java/org/apache/maven/lifecycle/internal/builder/multithreaded/ConcurrencyDependencyGraph.java

```diff
@@ -61,12 +61,12 @@ public int getNumberOfBuilds()
     /**
      * Gets all the builds that have no reactor-dependencies
      *
-     * @return A list of all the initial builds
+     * @return A set of all the initial builds
      */
 
-    public List<MavenProject> getRootSchedulableBuilds()
+    public Set<MavenProject> getRootSchedulableBuilds()
     {
-        List<MavenProject> result = new ArrayList<>();
+        Set<MavenProject> result = new HashSet<>();
```

Comment:
I think this should be `LinkedHashSet` to retain Insertion order just like `ArrayList` does.